effectly/src/components/Calendar/Calendar.tsx

```diff
@@ -0,0 +1,86 @@
+import { CalendarWrapper, EventsRows, NoEventsInfo } from "./Calendar.styles";
+
+import DatePicker from "./DatePicker/DatePicker";
+import Event from "./Event/Event";
+
+import useCalendarData from "../../hooks/Calendar/useCalendarData";
+import useDatePick from "../../hooks/Calendar/useDatePick";
+
+import { convertDate, filterEvents, getEventsMath } from "./helpers/events";
+import getRandomColor from "./helpers/getRandomColor";
+
+type CalendarProps = {
+  className: string;
+  userId: string;
+  children?: React.ReactNode;
+};
+
+const Calendar = ({ className, userId }: CalendarProps) => {
+  const { calData: calendarData } = useCalendarData(userId);
+  const {
+    calendarDay: visibleDay,
+    weekDayString: weekDay,
+    incrementDay,
+    decrementDay,
+  } = useDatePick();
+
+  const { sortedAndFilteredEvents: events } = filterEvents(
```

Comment:
Maybe would be good to add useCallback ?

Reply:
I've added useCallbacks to `filterEvents`'s internal functions as they were the only ones with dependencies + useMemos to internal functions that return values so they can be cached. Please check with [3d92612](https://github.com/wooczak/effectly/pull/8/commits/3d926123be704ef6bd4aa64bf76ec87eeae27ac8)